### DISCUSSION TITLE
RabbitMQ 3.8: basic.consume fails early on node boot before QQ leader is elected

### DISCUSSION THREAD

**Discussion by techblack** [rabbitmq/rabbitmq-server#10071] [Other] [answered]
### Describe the bug

Using single-node rabbitmq's Quorum Queues (qq), but leader not known appears. Command will be forwarded once leader is known.

- Rabbitmq version : 3.8.34
- one vhost
![image](https://github.com/rabbitmq/rabbitmq-server/assets/13714126/28300728-9d59-4faf-93e6-3ded8f3c5721)

### Reproduction steps

1. docker run `rabbitmq:3.8.34-management-alpine`
2. delcare a vhost and qq 
3. restart
...
 

### Expected behavior

qq always in leader

### Additional context

using `rabbitmqctl eval "ra:trigger_election('scanner_00000000000000000000000000000001')."`  can fix temporary it .
How should I completely resolve this issue?

**Comment by techblack** [comment]
This problem is not inevitable, but it is easy to occur on some machines.

**Comment by michaelklishin** [comment]
RabbitMQ 3.8 is [completely out of support](https://rabbitmq.com/versions.html). [Upgrading](https://rabbitmq.com/upgrade.html) is the only thing we can recommend.

**Comment by michaelklishin** [comment, marked as answer]
There is a hint in the stack trace. This happens in a `basic.consume` method handler, so delaying registering consumers on newly created QQs may serve as a temporary solution until you can upgrade to 3.12.10.
